agent/grpc/server_test.go

```diff
@@ -0,0 +1,28 @@
+package grpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/consul/agent/grpc/internal/testservice"
+)
+
+type simple struct {
+	name string
+	dc   string
+}
+
+func (s *simple) Flow(_ *testservice.Req, flow testservice.Simple_FlowServer) error {
+	for flow.Context().Err() == nil {
+		resp := &testservice.Resp{ServerName: "one", Datacenter: s.dc}
+		if err := flow.Send(resp); err != nil {
+			return err
+		}
+		time.Sleep(time.Millisecond)
```

Comment:
Probably fine but I wonder if this is slow enough to avoid being a hot loop or filling up buffers etc. if the test takes a while to run? Probably fine just seems like 1000 messages a second is a highish rate for a simple test!

Reply:
These tests run in under 10ms. I think anything slower would add unnecessary latency to the run time. The test reads a single message, then waits for this function to exit, so I think in practice this will only send 2-3 times.